### PR TITLE
[bugfix] Make `qml.equal` work with `qml.HilbertSchmidt`

### LINKED ISSUE #5536: [BUG] `qml.equal` does not work with `qml.HilbertSchmidt`
### Expected behavior

Equal ops are detected to be equal

### Actual behavior

Equal ops are reported to be different

### Additional information

The culprit is `v_tape` in the hyperparameters.

### Source code

```shell
u_tape = qml.tape.QuantumScript([qml.Hadamard(0)])
v_function = lambda params: qml.RZ(params[0], wires=1)
v_params = np.array([0.6])
op1 = qml.HilbertSchmidt(v_params, v_wires=[1], u_tape=u_tape, v_function=v_function)
op2 = qml.HilbertSchmidt(v_params, v_wires=[1], u_tape=u_tape, v_function=v_function)

qml.equal(op1, op2)
```

### Tracebacks

```shell
False
```

### System information

```shell
pl dev
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
`qml.HilbertSchmidt` has `QuantumScript` instances as hyperparameters, which are not compared correctly by the default `Operator` dispatch of `qml.equal`.

**Description of the Change:**
This PR adds a custom dispatch branch for `qml.HilbertSchmidt` to `qml.equal`, comparing the `QuantumScript` hyperparameters with additional `qml.equal` calls.

**Benefits:**
bug fix

**Possible Drawbacks:**

**Related GitHub Issues:**
fixes #5536 

[sc-61589]
[sc-49174]

### CODE REVIEW

File: tests/ops/functions/test_equal.py [hunk 1]
```diff
@@ -2071,3 +2073,153 @@ def test_non_equal_interfaces(self, op):
...
+    op7 = qml.HilbertSchmidt(v_params1, v_function=v_function3, v_wires=v_wires_alt, u_tape=u_tape1)
+
+    @pytest.mark.parametrize("op, other_op", [(op1, op1), (op2, op2), (op4, op4)])
+    def test_equality(self, op, other_op):
+        """Test that differing u_tapes are found."""
+        assert qml.equal(op, other_op) is True
+
+    # The second test case (incl op1_eps_tape) ensures that the kwargs of equal are
+    # passed to the tape comparisons correctly
+    @pytest.mark.parametrize("op, other_op", [(op1, op1_eps), (op1, op1_eps_tape)])
```
Comment (dwierichs) [inline tests/ops/functions/test_equal.py:2144]: `op1` and `op1_eps` differ by `1e-7` in their `params`.
`op1` and `op1_eps_tape` differ by `1e-7` in the parameters of their `u_tape`

File: tests/ops/functions/test_equal.py [hunk 2]
```diff
@@ -2071,3 +2073,153 @@ def test_non_equal_interfaces(self, op):
...
+        assert qml.equal(op, other_op) is False
+        # If the v_wires were different, so must have been the v_function and the
+        # resulting v_tape
+        new_other_op = deepcopy(other_op)
+        new_other_op.hyperparameters["v_function"] = op.hyperparameters["v_function"]
+        new_other_op.hyperparameters["v_tape"] = op.hyperparameters["v_tape"]
+        new_other_op.hyperparameters["v_wires"] = op.hyperparameters["v_wires"]
+        assert qml.equal(op, new_other_op) is True
+
+    @pytest.mark.parametrize("op, other_op", [(op5, op6), (op1_trainable, op1_untrainable)])
```
Comment (dwierichs) [inline tests/ops/functions/test_equal.py:2207]: `op5` and `op5` differ by the trainability of their `params`.
`op1_trainable` and `op1_untrainable` differ by the trainability of the parameters in `u_tape`.

File: tests/ops/functions/test_equal.py [hunk 3]
```diff
@@ -2071,3 +2073,153 @@ def test_non_equal_interfaces(self, op):
...
+    def test_trainability(self, op, other_op):
+        """Test that differing trainabilities are found."""
+        assert qml.equal(op, other_op) is False
+        assert qml.equal(op, other_op, check_trainability=False) is True
+
+    @pytest.mark.parametrize("op, other_op", [(op1, op6), (op1, op1_untrainable)])
```
Comment (dwierichs) [inline tests/ops/functions/test_equal.py:2213]: `op1` and `op6` differ by the interface of their `params`.
`op1` and `op1_untrainable` differ by the interface of the parameters in `u_tape`.

File: tests/ops/functions/test_equal.py [hunk 4]
```diff
@@ -2071,3 +2073,153 @@ def test_non_equal_interfaces(self, op):
...
+    def test_interface(self, op, other_op):
+        """Test that differing interfaces are found."""
+        assert qml.equal(op, other_op) is False
+        assert qml.equal(op, other_op, check_interface=False) is True
+
+    @pytest.mark.parametrize("op, other_op", [(op1, op5), (op1, op1_trainable)])
```
Comment (dwierichs) [inline tests/ops/functions/test_equal.py:2219]: `op5` and `op5` differ by the trainability &interface of their `params`.
`op1_trainable` and `op1_untrainable` differ by the trainability &interface of the parameters in `u_tape`.

File: tests/transforms/test_hamiltonian_expand.py [hunk 5]
```diff
@@ -345,6 +345,7 @@ def test_processing_function_shot_vectors_broadcasting(self, H, expected, groupi
         """Tests that the processing function works with shot vectors, parameter broadcasting,
         and grouping with different number of coefficients in each group"""
 
+        np.random.seed(824)
```
Comment (dwierichs) [inline tests/transforms/test_hamiltonian_expand.py:348]: This was randomly failing.

**Review by albi3ro — APPROVED** [review]

**Review by astralcai — APPROVED** [review]

**Pull request merged**